runtime/src/main/java/org/corfudb/runtime/view/ManagementView.java

```diff
@@ -162,13 +168,21 @@ private ClusterStatus getSequencerServersClusterHealth(Layout layout,
      */
     private ClusterStatus getLogUnitServersClusterHealth(Layout layout,
                                                          Set<String> peerResponsiveNodes) {
+        // logUnitRedundancyStatus marks the cluster as degraded if any of the nodes is performing
+        // stateTransfer and is in process of achieving full redundancy.
+        ClusterStatus logUnitRedundancyStatus = peerResponsiveNodes.stream()
+                .anyMatch(s -> getLogUnitNodeStatusInLayout(layout, s) == NodeStatus.DB_SYNCING)
+                ? ClusterStatus.DEGRADED : ClusterStatus.STABLE;
         // Check the availability of the log servers in all segments as reads to all addresses
         // should be accessible.
-        return layout.getSegments().stream()
+        ClusterStatus logunitClusterStatus = layout.getSegments().stream()
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Call "Optional#isPresent()" before accessing the value. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3655)